mypy/build.py

```diff
@@ -1587,7 +1598,9 @@ def __init__(self,
                 self.interface_hash = self.meta.interface_hash
                 self.meta_source_hash = self.meta.hash
         self.add_ancestors()
+        t0 = time.time()
         self.meta = validate_meta(self.meta, self.id, self.path, self.ignore_all, manager)
+        self.manager.add_stats(validate_meta_time=time.time() - t0)
```

Comment:
I would recommend also checking if the special case below on line 1629 takes some time (just in case).

Reply:
That only fires when not using a fine-grained cache, so I don't think it's a possibility for the cases we care about

Reply:
Ah yes, indeed, it will be taken care of during the initial update. I remember it caused big troubles before, but it was in normal incremental mode.